### ISSUE TITLE
Make the base 83 encoding optional.

### ISSUE THREAD

**Issue by iomintz** [woltapp/blurhash#93]
Suppose I want to store 80,000 pre-blurred images in my database. `LjIY5?00?bIUofWBWBM{WBofWBj[` is a sample 4×3 blurHash, and is 28 bytes (224 bits) long. Suppose I use a fixed-width data type in my database, so I don't need the first character of the string which encodes the length. If we remove the first character and encode the same blurHash in its equivalent raw bytes, we need only ~10 bytes (73 bits) to encode the hash, saving 1.5 MB on disk for 80,000 images. That may seem like a micro-optimization, especially in an RDBMS which may duplicate data in indexes. You could say I'm just allergic to plaintext formats when a more compact representation would do, and you'd be right :) I still think there's value in this approach though.

**Comment by jagerman** [comment]
There definitely is merit in this: base83 is imposing the (severe) limitations of JSON binary encoding onto the user of the library, when the user of the library might be perfectly capable of representing binary data as binary data (as in the OP's case).  Helpers for JSON/HTML compatibility are fine, but right now they are forced on everyone rather than only being forced on people who are themselves forced to deal with JSON's limitations.

**Comment by DagAgren** [comment]
The raw data of BlurHash is not binary data. The encoding is adapted directly to the internal data format. For instance, most of the string is AC components, which are RGB triplets where each value has 19 possible values. 19*19*19 just fits into 83*83. A binary encoding of this would be less efficient.

**Comment by jagerman** [comment]
> A binary encoding of this would be less efficient.

No, that's not how binary encoding works.  A *bad* binary encoding could be less efficient, indeed, but by insisting on a base-83 storage, you are choosing to store slightly less than 13 bits of data in 16 bits of storage.  That is less efficient, not more efficient.  Or to put it another way: every possible 28 bytes base-83 string (such as the 28-byte one @ioistired mentioned) can be represented in 23 bytes of base-256 storage.
